src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/ConstructorSymbolReferenceFinder.cs

```diff
@@ -146,5 +147,55 @@ protected override bool CanFind(IMethodSymbol symbol)
                 ? FindReferencesInDocumentUsingIdentifierAsync(symbol, simpleName, document, semanticModel, cancellationToken)
                 : new ValueTask<ImmutableArray<FinderLocation>>(ImmutableArray<FinderLocation>.Empty);
         }
+
+        private Task<ImmutableArray<FinderLocation>> FindReferencesInImplicitObjectCreationExpressionAsync(
+            IMethodSymbol symbol,
+            Document document,
+            SemanticModel semanticModel,
+            CancellationToken cancellationToken)
+        {
+            // Only check `new (...)` calls that supply enough arguments to match all the required parameters for the constructor.
+            var minimumArgumentCount = symbol.Parameters.Count(p => !p.IsOptional && !p.IsParams);
+            var maximumArgumentCount = symbol.Parameters.Length > 0 && symbol.Parameters.Last().IsParams
+                ? int.MaxValue
+                : symbol.Parameters.Length;
+
+            var exactArgumentCount = symbol.Parameters.Any(p => p.IsOptional || p.IsParams)
+                ? -1
+                : symbol.Parameters.Length;
+
+            return FindReferencesInDocumentAsync(symbol, document, IsRelevantDocument, CollectMatchingReferences, cancellationToken);
+
+            static bool IsRelevantDocument(SyntaxTreeIndex syntaxTreeInfo)
+                => syntaxTreeInfo.ContainsImplicitObjectCreation;
+
+            void CollectMatchingReferences(
+                ISymbol originalUnreducedSymbolDefinition, SyntaxNode node,
+                ISyntaxFactsService syntaxFacts, ISemanticFactsService semanticFacts,
+                ArrayBuilder<FinderLocation> locations)
+            {
+                if (!syntaxFacts.IsImplicitObjectCreationExpression(node))
```

Comment:
@CyrusNajmabadi There is another instance method, "IsImplicitObjectCreation". Why we need to have an extension that does exactly the same thing?